File: 0242-valid-anagram/0242-valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s)!=len(t):
            return False
        
        str1 ={}
        str2 ={}

        for i in range(0,len(s)):
            if s[i] in str1:
                str1[s[i]]+=1
            else:
                str1[s[i]]=1
            if t[i] in str2:
                str2[t[i]]+=1
            else:
                str2[t[i]]=1

        if(str1==str2):
            return True
        else:
            return False

            
            

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {

        if(s.length() != t.length()){
            return false;
        }

        int[] ch1 = new int[26];
        int[] ch2 = new int[26];

        for(int i = 0; i<s.length();i++){

            int x = (int) s.charAt(i) - (int) 'a';
            int y = (int) t.charAt(i) - (int) 'a';

            ch1[x] = ch1[x]+1;
            ch2[y] = ch2[y]+1;
        }

        if(Arrays.equals(ch1,ch2)){
            return true;
        }
        return false;



    }
}